### PR TITLE
Update to 3.1.1, add python 3 builds

### PR DESCRIPTION
**Pull request by sdvillal**
This enables us to create python 3 builds of caffe. We have test it in linux, hopefully it will work in the other systems as well (I suspect it is going to fail in windows, as *single-version-externally-managed* seems not to work anymore).

I have some simple questions about best practices (see patch comments):
  1. Is there any better URL we can use to get the package?
  2. [Attending to this](http://conda-forge.github.io/docs/meta.html#single-verion-externally-managed) I'm removing by hand any ```.egg-info``` file in the package. I do not actually know if this is correct (alghough I strongly suspect it is, and we are already using this package in our environments). If it is correct, could it be worth completing conda-forge docs to add this little trick?
  3. Can the maintainer list contain extra information, like email address or affiliation?

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,25 +1,30 @@
...
 package:
   name: python-gflags
   version: {{ version }}
 
 source:
   fn: python-gflags-{{ version }}.tar.gz
-  url: https://pypi.python.org/packages/source/p/python-gflags/python-gflags-{{ version }}.tar.gz
-  md5: 23c9a793959a54971b1f094b0c6d03b1
+  # ugly url, anything better?
+  url: https://pypi.python.org/packages/ea/30/b8469c0d1837ce58fe3706e1f7169cbf6ca1fb87d1f84cece5182b67cb0b/python-gflags-3.1.1.tar.gz
```
Comment (jakirkham) [inline recipe/meta.yaml]: Please look at the [`example`]( https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml ) recipe and update accordingly.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,25 +1,30 @@
...
 
 build:
   number: 1
-  skip: True  # [py3k]
-  script: python setup.py install --single-version-externally-managed --record record.txt
+  script:
+    # Does not work anymore
+    # - python setup.py install --single-version-externally-managed --record record.txt
+    - python setup.py install
+    - rm -Rf $PREFIX/*.egg-info
```
Comment (jakirkham) [inline recipe/meta.yaml]: Why doesn't `--single-version-externally-managed --record record.txt` work? Also why do we need to remove an `.egg-info` directory?

Comment (jakirkham) [reply recipe/meta.yaml]: Ah, they stopped using `setuptools`. So dropping the args is fine. Still contend there is nothing that we need to do with `.egg-info`.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,25 +1,30 @@
...
 
 requirements:
   build:
     - python
     - setuptools
```
Comment (jakirkham) [inline recipe/meta.yaml]: Please drop `setuptools` as they are not using this any more.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,25 +1,30 @@
...
 
   run:
+    - six
```
Comment (jakirkham) [inline recipe/meta.yaml]: Thanks for adding. Strange that they think that they can manage dependencies with `distutils`.

xref: https://github.com/google/python-gflags/blob/3.1.1/setup.py#L43
xref: https://github.com/google/python-gflags/commit/21f866832a20c6c7cc53930a39953afab0e8eda2#commitcomment-21106396

**Comment by sdvillal** [comment]
Thanks for the comments, I have incorporated all the changes.

I'm surprised that the windows build, without changing *bld.bat*, still works.

**Comment by jakirkham** [comment]
Oh yeah, thanks for the reminder. `conda-build` didn't use to use `script` on Windows. So we had to add `bld.bat`, but `conda-build` was fixed a while back. So I think it is ignoring it now as it can use `script`. Could you please remove `bld.bat` while we are at it?

**Comment by sdvillal** [comment]
Done and ready to merge as soon as we get the green tick.

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render.

**Comment by jakirkham** [comment]
Thanks @sdvillal. Sorry this one got lost for so long. 😞

**Pull request merged**